burn-tensor/src/tensor/api/base.rs

```diff
@@ -401,43 +402,176 @@ where
 ///
 /// This is an internal trait, use the public API provided by [tensor struct](Tensor).
 pub trait BasicOps<B: Backend>: TensorKind<B> {
+    /// The type of the tensor elements.
```

Comment:
The doc is nice, but this is an internal trait to call different backend functions depending on the data type with static dispatch. We can't make it private, but users are not supposed to import or use it. So in the the doc, I would refer to the right tensor function (e.g `Tensor::reshape`). I'm not sure how we can make it more obvious that it is an internal trait, do you have any suggestions?